Additional tests and fixes

scottmcdonnell opened the pull request:
I had a few issues with the 1.1.0 sync.
- tests were not passing 100% for me
- user names with spaces failing validation
- users with no last name failing validation
- concept task type missing - gazu.task.all_task_types_for_project(kitsu_project_id) =>  gazu.task.all_task_types()
- concept where data =None was causing `calculate_end_frame` to fail
- added fixtures to update the studio_settings for tests
- added tests for entity Person
- added tests for entity Edit
- added tests for entity Concept
- user isAdmin and isManager were not being set for me
- added the the accessGroup to the project as users were not being assigned without it
- delete projects setting bug

Lastly some of new bits were using the Ayon api via httpx to save users/projects etc but this is mixed with the existing `server/kitsu/utils.py` which uses Postgres for gets and Entity models to save and update. This I found really confusing, it should be all one or the other. It didn't make much sense to me for Ayon (in an ayon api endpoint) to be calling itself through the api when the models are available and the postgres connection is already initialised, it seems an unnecessary overhead,  so I converted the api calls to model calls in utils.py to make it consistent. 

I saw afterwards that this was discussed on discord a bit, and the preference was API, but I think it should be consistent throughout utils.py - so not sure what the team thinks on this one!

![Screenshot 2024-02-25 at 15 00 48](https://github.com/ynput/ayon-kitsu/assets/1951220/244ae5b5-c2a8-4eec-bcf0-06d78042204b)

Commit:
improve test readme

Commit:
fixtures for users and studio_settings, initial test for kitsu Person

Commit:
return users from push for testing

Commit:
bugfix - handle kitsu person name with spaces or special characters

Commit:
Merge remote-tracking branch 'ayon/main' into enhancement/fully_working_sync

Commit:
do not set password if blank

Commit:
UserEntity instead of httpx client call for user update.

Commit:
fix mock data and tests for person entities

Commit:
handle change of user name

Commit:
bugfix - only calculate endframe if not explicitly set.

Commit:
ProjectEntity instead of httpx, project anatomy updating - all tests passing

Commit:
bugfix -  settings.delete_ayon_projects.enabled =>  settings.sync_settings.delete_projects - tests for deleting project with setting enabled and disabled

Commit:
optional param for mock and test improvements

Commit:
fix for usernames with no first or last name failing ayon validation

Commit:
try catch for entities push so if they fail the entity is logged and we skip onto the next entity rather than stopping.

Commit:
test cleanup and version bump

Commit:
bugfix for isAdmin and isManager to set correctly. add project to access groups

Commit:
try and catch for remove entity so problems can be logged and skipped

Commit:
bug fix for concepts having no data

Commit:
tests for edits and concepts

Commit:
bugfix for concept task type missing as not on the project

Commit:
init_pairing test fix

Commit:
bugfix support names with unicode characters - move any formatting utils into its own module for easy testing.

Commit:
do not sync Kitsu bots

Commit:
remove ayon_server_url as unnecessary

Commit:
support for LauncherAction - Show In Kitsu

Commit:
working launcher_show_in_kitsu

(cherry picked from commit b1da619020ec2b4d5ed1b2d6b36d4303bae052c4)

scottmcdonnell commented:
added support for LauncherAction https://github.com/ynput/ayon-kitsu/issues/36

Commit:
fix all tests to support kitsuType on folders

Commit:
add version for direct mounting ayon-kitsu/server in ayon-docker

EmberLightVFX commented:
Would it be possible to change the arrow-icon to the Kitsu icon for the "Show in Kitsu" button?

iLLiCiTiT commented:
Hello, could you update the changes? Could you also split the PR into more PRs? There is too much changes for review (smaller PRs -> faster review).

scottmcdonnell commented:
> Hello, could you update the changes? Could you also split the PR into more PRs? There is too much changes for review (smaller PRs -> faster review).

Thanks for taking a look. I can give it a go breaking it down into different PRs. 
Before I go ahead can I ask you to review one big structural change in this PR to see if I should proceed down this path?

on the server side there is a mix between using EntityModels and api calls in the current code. This PR replaces any api with Entity in order to make it consistent.  From above:

> Lastly some of new bits were using the Ayon api via httpx to save users/projects etc but this is mixed with the existing server/kitsu/utils.py which uses Postgres for gets and Entity models to save and update. This I found really confusing, it should be all one or the other. It didn't make much sense to me for Ayon (in an ayon api endpoint) to be calling itself through the api when the models are available and the postgres connection is already initialised, it seems an unnecessary overhead, so I converted the api calls to model calls in utils.py to make it consistent.

For example:
https://github.com/ynput/ayon-kitsu/pull/35/commits/7559d8b5c3365c9e5380b75240e3558160f293c8

iLLiCiTiT commented:
> Before I go ahead can I ask you to review one big structural change in this PR to see if I should proceed down this path?

The issue is that different changes should be reviewed by different people, that's why I asked for split. It is hard to tell which change is related to what.

scottmcdonnell commented:
> > Before I go ahead can I ask you to review one big structural change in this PR to see if I should proceed down this path?
> 
> The issue is that different changes should be reviewed by different people, that's why I asked for split. It is hard to tell which change is related to what.
> 
> There are some small bugfixes and big feature changes next to it (multiple features). The features usually needs discussion or may not be merged because of different opinion, but the bugfixes can be merged right away.

Fair enough, I get that. Would it be possible to get feedback on the replacement of API for Entities please?
https://github.com/ynput/ayon-kitsu/commit/7559d8b5c3365c9e5380b75240e3558160f293c8

If that is given the thumbs up or down I will know how to proceed with the PRs.

iLLiCiTiT commented:
> Fair enough, I get that. Would it be possible to get feedback on the replacement of API for Entities please?
https://github.com/ynput/ayon-kitsu/commit/7559d8b5c3365c9e5380b75240e3558160f293c8

@martastain

m-u-r-p-h-y commented:
First of all thank you @scottmcdonnell  for your contribution! To push it through, we really need to split this massive PR to smaller pieces as it is not approvable in the current state. 

each bulleted item in your description should be a separate PR honestly.

scottmcdonnell commented:
> First of all thank you @scottmcdonnell for your contribution! To push it through, we really need to split this massive PR to smaller pieces as it is not approvable in the current state.
> 
> each bulleted item in your description should be a separate PR honestly.

No problem, I will break into smaller PRs. I am on a few big projects at the moment but will come back to this as soon as I can.

BigRoy commented:
Would be great to get some 🔥 progress into this one.
Someone else just hit what seems like something that this PR may solve? See on [Ynput discord here](https://discord.com/channels/517362899170230292/563751989075378201/1247281052460777573).

scottmcdonnell commented:
> Would be great to get some 🔥 progress into this one. Someone else just hit what seems like something that this PR may solve? See on [Ynput discord here](https://discord.com/channels/517362899170230292/563751989075378201/1247281052460777573).

Sorry was very snowed under. PRs started. Can I make PRs dependant on each other? The testing requires fixtures and set up that is common to each PR and some changes have larger dependencies.

MustafaJafar commented:
Hello, 
I think this PR is big, would be better to break it down into smaller PRs ? 
e.g. could we have separate PR for the launcher action ?

BigRoy commented:
This PR is massive, with commented out code in between feels unfinished and comes with quite a few conflicts that need resolving. With this being pending for so long - I'll close it. If there's worth in some of the changes someone will have to re-open and continue the efforts with only the needed changes.
